client/app/bundles/course/assessment/submission/actions/index.js

```diff
@@ -484,15 +485,70 @@ export function saveGrade(submissionId, grades, exp, published) {
   };
 
   return (dispatch) => {
-    dispatch({ type: actionTypes.SAVE_GRADE_REQUEST });
+    dispatch({ type: actionTypes.SAVE_ALL_GRADE_REQUEST });
 
     return CourseAPI.assessment.submissions
       .update(submissionId, payload)
       .then((response) => response.data)
       .then((data) => {
-        dispatch({ type: actionTypes.SAVE_GRADE_SUCCESS, payload: data });
+        dispatch({ type: actionTypes.SAVE_ALL_GRADE_SUCCESS, payload: data });
```

Comment:
You are renaming `SAVE_GRADE_REQUEST`, `SAVE_GRADE_SUCCESS` and `SAVE_GRADE_FAILURE`. Have you checked other places that uses the same action names?

Reply:
yes, thus far, in the other places (other than actually saving the grade, for example in annotation, explanation, etc.), I assume both SAVE_ALL_GRADE_SUCCESS and SAVE_GRADE_SUCCESS have the same effect on the states.

Though, this warrants the further inspection (thus far I didn't see any difference in terms of effect) on whether they are actually supposed to have the same effect. I will check on it further later.

Reply:
What I mean is that, previously, saveGrade uses `SAVE_GRADE_SUCCESS` . This action also exists in multiple reducers (ie annotations, answers, attachment, etc). Once you rename to `SAVE_ALL_GRADE_SUCCESS`, why `SAVE_ALL_GRADE_SUCCESS` is not added to the other reducers as well?

Reply:
oh I see. yes regarding that one, it's the mistake from my part that I didn't add `SAVE_ALL_GRADE_SUCCESS` to all the places (mainly because I was focused on saving grades, unaware that the other reducers is using this one). In the current commit I have just pushed (it's unfinished), I have added that to other reducers as well..

but it might need some further inspection, which I will do later